src/osp/Universe.cpp

```diff
@@ -49,27 +53,61 @@ void Universe::sat_remove(Satellite sat)
     m_registry.destroy(sat);
 }
 
-Vector3s Universe::sat_calc_pos(Satellite referenceFrame, Satellite target) const
+Vector3g Universe::sat_calc_pos(Satellite referenceFrame, Satellite target) const
 {
-    auto const view = m_registry.view<const UCompTransformTraj>();
+    auto const viewInCoord = m_registry.view<const UCompInCoordspace>();
+    auto const viewCoordIndex = m_registry.view<const UCompCoordspaceIndex>();
 
     // TODO: maybe do some checks to make sure they have the components
-    auto const &framePosTraj = view.get<const UCompTransformTraj>(referenceFrame);
-    auto const &targetPosTraj = view.get<const UCompTransformTraj>(target);
+    auto const &frameInCoord
+            = viewInCoord.get<const UCompInCoordspace>(referenceFrame);
+    auto const &targetInCoord
+            = viewInCoord.get<const UCompInCoordspace>(target);
 
-    if (framePosTraj.m_parent == targetPosTraj.m_parent)
+    auto const &frameCoordIndex
+            = viewCoordIndex.get<const UCompCoordspaceIndex>(referenceFrame);
+    auto const &targetCoordIndex
+            = viewCoordIndex.get<const UCompCoordspaceIndex>(target);
+
+    if (frameInCoord.m_coordSpace == targetInCoord.m_coordSpace)
     {
-        // Same parent, easy calculation
-        return targetPosTraj.m_position - framePosTraj.m_position;
+        // Both Satellites are in the same coordinate system
+
+        std::optional<CoordinateSpace> const &rSpace
+                = m_coordSpaces.at(frameInCoord.m_coordSpace);
+
```

Comment:
Need to verify / assert that the optional has a value before you use it in the line below.

Reply:
I'm planning on making it guaranteed for it to never be possible for `frameInCoord.m_coordSpace` here to point to an invalid coordinate space.

Reply:
Gotcha. Then an assert would be the right call.